Include/internal/pycore_bitutils.h

```diff
@@ -131,8 +131,46 @@ _Py_popcount32(uint32_t x)
 }
 
 
+// Return the index of the most significant 1 bit in 'x'. This is the smallest
+// integer k such that x < 2**k. Equivalent to floor(log2(x)) + 1 for x != 0.
+static inline int
+_Py_bit_length(unsigned long x)
+{
+#if (defined(__clang__) || defined(__GNUC__))
+    if (x != 0) {
+        // __builtin_clzl() is available since GCC 3.4.
+        // Undefined behavior for x == 0.
+        return sizeof(unsigned long) * 8 - __builtin_clzl(x);
```

Comment:
Maybe cast explicitly to int to control which types are used.

```suggestion
        return (int)sizeof(unsigned long) * 8 - __builtin_clzl(x);
```